### PR TITLE
[JENKINS-46094] Fix SSH Clone trait using agent credentials

### PR DESCRIPTION
**Pull request by bitwiseman**
@abayer @rsandell @dwnusbaum

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMBuilder.java [hunk 1]
```diff
@@ -181,8 +187,29 @@ public final String repositoryUrl(String owner, String repo) {
      */
     @NonNull
     public final RepositoryUriResolver uriResolver() {
-        String credentialsId = credentialsId();
-        return uriResolver(context, apiUri, credentialsId);
+        return uriResolver;
+    }
+
+        /**
```
Comment (jsoref) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMBuilder.java]: Odd indentation

File: src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMBuilder.java [hunk 2]
```diff
@@ -62,11 +62,11 @@
     /**
      * Singleton instance of {@link HttpsRepositoryUriResolver}.
      */
-    private static final HttpsRepositoryUriResolver HTTPS = new HttpsRepositoryUriResolver();
+    public static final HttpsRepositoryUriResolver HTTPS = new HttpsRepositoryUriResolver();
```
Comment (rsandell) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMBuilder.java]: It is probably enough to make these package protected.

Comment (rsandell) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMBuilder.java]:
```suggestion
    static final HttpsRepositoryUriResolver HTTPS = new HttpsRepositoryUriResolver();
```

File: src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMBuilder.java [hunk 3]
```diff
@@ -62,11 +62,11 @@
...
     /**
      * Singleton instance of {@link SshRepositoryUriResolver}.
      */
-    private static final SshRepositoryUriResolver SSH = new SshRepositoryUriResolver();
+    public static final SshRepositoryUriResolver SSH = new SshRepositoryUriResolver();
```
Comment (rsandell) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMBuilder.java]:
```suggestion
    static final SshRepositoryUriResolver SSH = new SshRepositoryUriResolver();
```

File: src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMBuilder.java (hunk 1 above)
Comment (bitwiseman) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMBuilder.java]:
```suggestion
    /**
```

**Pull request merged**